From 3957a02cb1ea384b16d0f6e93e837f77f5e7781c Mon Sep 17 00:00:00 2001
From: huangxin <huangxin@LServer.(none)>
Date: Wed, 16 Nov 2011 15:35:33 +0800
Subject: [PATCH 326/944] [sun4i]:dma,hrtimer,ace,use the codec style to
 modify the dma ace hrtimer style and change the
 hrtimer config the same as linux2.6.36

---
 arch/arm/mach-sun4i/clock/aw_clocksrc.c | 67 ++++++++++++++-------------------
 arch/arm/mach-sun4i/clock/aw_clocksrc.h | 30 +++++++--------
 arch/arm/mach-sun4i/dma/dma.c           | 33 +++++++---------
 arch/arm/mach-sun4i/dma/dma_15.c        | 31 +++++++--------
 arch/arm/mach-sun4i/dma/dma_regs.h      | 29 +++++++-------
 arch/arm/mach-sun4i/include/mach/dma.h  | 29 +++++++-------
 drivers/media/audio/sun4i_dev_ace.c     | 19 +++++-----
 7 files changed, 104 insertions(+), 134 deletions(-)

diff --git a/arch/arm/mach-sun4i/clock/aw_clocksrc.c b/arch/arm/mach-sun4i/clock/aw_clocksrc.c
index f229de3..370be91 100644
--- a/arch/arm/mach-sun4i/clock/aw_clocksrc.c
+++ b/arch/arm/mach-sun4i/clock/aw_clocksrc.c
@@ -1,21 +1,17 @@
 /*
-*********************************************************************************************************
-*                                                    LINUX-KERNEL
-*                                        AllWinner Linux Platform Develop Kits
-*                                                   Kernel Module
-*
-*                                    (c) Copyright 2006-2011, kevin.z China
-*                                             All Rights Reserved
-*
-* File    : aw_clocksrc.c
-* By      : kevin.z
-* Version : v1.0
-* Date    : 2011-5-10 10:55
-* Descript: clock source driver for aw chips
-* Update  : date                auther      ver     notes
-*
-*********************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\clock\aw_clocksrc.c
+ * (C) Copyright 2010-2015
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * kevin.z
+ *
+ * core header file for Lichee Linux BSP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 #include <mach/hardware.h>
 #include <mach/platform.h>
 #include <linux/init.h>
@@ -39,8 +35,9 @@
 #endif
 
 static cycle_t aw_clksrc_read(struct clocksource *cs);
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static irqreturn_t aw_clkevt_irq(int irq, void *handle);
+static spinlock_t timer1_spin_lock;
 static void aw_set_clkevt_mode(enum clock_event_mode mode, struct clock_event_device *dev);
 static int aw_set_next_clkevt(unsigned long delta, struct clock_event_device *dev);
 #endif
@@ -60,7 +57,7 @@ static struct clocksource aw_clocksrc =
     .flags = CLOCK_SOURCE_IS_CONTINUOUS,
 };
 
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static struct clock_event_device aw_clock_event =
 {
     .name = "aw clock event device",
@@ -140,7 +137,7 @@ static cycle_t aw_clksrc_read(struct clocksource *cs)
 *
 *********************************************************************************************************
 */
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static void aw_set_clkevt_mode(enum clock_event_mode mode, struct clock_event_device *dev)
 {
     CLKSRC_DBG("aw_set_clkevt_mode:%u\n", mode);
@@ -198,32 +195,26 @@ static void aw_set_clkevt_mode(enum clock_event_mode mode, struct clock_event_de
 *
 *********************************************************************************************************
 */
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static int aw_set_next_clkevt(unsigned long delta, struct clock_event_device *dev)
 {
+	unsigned long flags;
     CLKSRC_DBG("aw_set_next_clkevt: %u\n", (unsigned int)delta);
 
-    /* time value timer must larger than 50 cycles at least, suggested by david 2011-5-25 11:41 */
-    if(delta < 50)
-    {
-        delta = 50;
-    }
-
+	spin_lock_irqsave(&timer1_spin_lock, flags);
     /* disable timer and clear pending first    */
     TMR_REG_TMR1_CTL &= ~(1<<0);
     /* wait hardware synchronization, 2 cycles of the hardware work clock at least  */
-    __delay(50);
+    udelay(1);
 
     /* set timer intervalue         */
     TMR_REG_TMR1_INTV = delta;
     /* reload the timer intervalue  */
     TMR_REG_TMR1_CTL |= (1<<1);
-    /* wait hardware synchronization, 2 cycles of the hardware work clock at least  */
-    __delay(50);
 
     /* enable timer */
     TMR_REG_TMR1_CTL |= (1<<0);
-
+    spin_unlock_irqrestore(&timer1_spin_lock, flags);
     return 0;
 }
 #endif
@@ -245,16 +236,16 @@ static int aw_set_next_clkevt(unsigned long delta, struct clock_event_device *de
 *
 *********************************************************************************************************
 */
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static irqreturn_t aw_clkevt_irq(int irq, void *handle)
 {
     if(TMR_REG_IRQ_STAT & (1<<1))
     {
         CLKSRC_DBG("aw_clkevt_irq!\n");
-        /* clock event interrupt handled */
-        aw_clock_event.event_handler(&aw_clock_event);
         /* clear pending */
         TMR_REG_IRQ_STAT = (1<<1);
+        /* clock event interrupt handled */
+        aw_clock_event.event_handler(&aw_clock_event);
 
         return IRQ_HANDLED;
     }
@@ -305,7 +296,6 @@ static int __init aw_clksrc_init(void)
     return 0;
 }
 
-
 /*
 *********************************************************************************************************
 *                           aw_clkevt_init
@@ -322,7 +312,7 @@ static int __init aw_clksrc_init(void)
 *
 *********************************************************************************************************
 */
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 static int __init aw_clkevt_init(void)
 {
     /* register clock event irq     */
@@ -349,7 +339,8 @@ static int __init aw_clkevt_init(void)
     CLKSRC_DBG("register all-winners clock event device!\n");
 	aw_clock_event.mult = div_sc(AW_HPET_CLOCK_EVENT_HZ, NSEC_PER_SEC, aw_clock_event.shift);
 	aw_clock_event.max_delta_ns = clockevent_delta2ns((0x80000000), &aw_clock_event);
-	aw_clock_event.min_delta_ns = clockevent_delta2ns(1, &aw_clock_event) + 3000;
+	/* time value timer must larger than 50 cycles at least, suggested by david 2011-5-25 11:41 */
+	aw_clock_event.min_delta_ns = clockevent_delta2ns(1, &aw_clock_event) + 4000;
 	aw_clock_event.cpumask = cpumask_of(0);
     clockevents_register_device(&aw_clock_event);
 
@@ -358,6 +349,6 @@ static int __init aw_clkevt_init(void)
 #endif
 
 arch_initcall(aw_clksrc_init);
-#if 0
+#ifdef CONFIG_HIGH_RES_TIMERS
 arch_initcall(aw_clkevt_init);
 #endif
diff --git a/arch/arm/mach-sun4i/clock/aw_clocksrc.h b/arch/arm/mach-sun4i/clock/aw_clocksrc.h
index ecf0208..3e7feaa 100644
--- a/arch/arm/mach-sun4i/clock/aw_clocksrc.h
+++ b/arch/arm/mach-sun4i/clock/aw_clocksrc.h
@@ -1,21 +1,17 @@
 /*
-*********************************************************************************************************
-*                                                    LINUX-KERNEL
-*                                        AllWinner Linux Platform Develop Kits
-*                                                   Kernel Module
-*
-*                                    (c) Copyright 2006-2011, kevin.z China
-*                                             All Rights Reserved
-*
-* File    : aw_clocksrc.h
-* By      : kevin.z
-* Version : v1.0
-* Date    : 2011-5-10 13:32
-* Descript: head file for clock source on aw chips;
-* Update  : date                auther      ver     notes
-*
-*********************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\clock\aw_clocksrc.h
+ * (C) Copyright 2010-2015
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * kevin.z
+ *
+ * core header file for Lichee Linux BSP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 #ifndef __AW_CLOCKSRC_H__
 #define __AW_CLOCKSRC_H__
 
diff --git a/arch/arm/mach-sun4i/dma/dma.c b/arch/arm/mach-sun4i/dma/dma.c
index dca6bfd..fbb0b63 100644
--- a/arch/arm/mach-sun4i/dma/dma.c
+++ b/arch/arm/mach-sun4i/dma/dma.c
@@ -1,24 +1,17 @@
 /*
-**************************************************************************************************************
-*											         eLDK
-*						            the Easy Portable/Player Develop Kits
-*									           desktop system
-*
-*						        	 (c) Copyright 2009-2012, ,HUANGXIN China
-*											 All Rights Reserved
-*
-* File    	: dma.c
-* By      	: HUANGXIN
-* Func		:
-* Version	: v1.0
-* ============================================================================================================
-* 2011-5-9 10:37:48  HUANGXIN create this file, implements the fundemental interface;
-* 2011-10-28         benn port this to kernel-3.0
-*                    1) suspend/resume was moved to dma_15.c
-*                    2) use platform_device to replace sysdev
-*                    3) make dma_regs.h to private
-**************************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\dma\dma.c
+ * (C) Copyright 2007-2011
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * huangxin <huangxin@allwinnertech.com>
+ *
+ * some simple description for this code
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 //#define DEBUG
 
 #include <linux/module.h>
diff --git a/arch/arm/mach-sun4i/dma/dma_15.c b/arch/arm/mach-sun4i/dma/dma_15.c
index 624fdbd..c9e4690 100644
--- a/arch/arm/mach-sun4i/dma/dma_15.c
+++ b/arch/arm/mach-sun4i/dma/dma_15.c
@@ -1,22 +1,17 @@
 /*
-**************************************************************************************************************
-*											         eLDK
-*						            the Easy Portable/Player Develop Kits
-*									           desktop system
-*
-*						        	 (c) Copyright 2009-2012, ,HUANGXIN China
-*											 All Rights Reserved
-*
-* File    	: dma_15.h
-* By      	: HUANGXIN
-* Func		:
-* Version	: v1.0
-* ============================================================================================================
-* 2011-5-9 10:36:54  HUANGXIN create this file, implements the fundemental interface;
-* 2011-10-28         benn port to kernel-3.0
-*                    1) FIXME: to add sw_dmac_suspend/sw_dmac_resume implementation
-**************************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\dma\dma_15.c
+ * (C) Copyright 2007-2011
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * huangxin <huangxin@allwinnertech.com>
+ *
+ * some simple description for this code
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 #include <linux/kernel.h>
 #include <linux/init.h>
 #include <linux/sysdev.h>
diff --git a/arch/arm/mach-sun4i/dma/dma_regs.h b/arch/arm/mach-sun4i/dma/dma_regs.h
index 0b9963b..9eb3894 100644
--- a/arch/arm/mach-sun4i/dma/dma_regs.h
+++ b/arch/arm/mach-sun4i/dma/dma_regs.h
@@ -1,20 +1,17 @@
 /*
-**************************************************************************************************************
-*											         eLDK
-*						            the Easy Portable/Player Develop Kits
-*									           desktop system
-*
-*						        	 (c) Copyright 2009-2012, ,HUANGXIN China
-*											 All Rights Reserved
-*
-* File    	: dma_regs.h
-* By      	: HUANGXIN
-* Func		:
-* Version	: v1.0
-* ============================================================================================================
-* 2011-5-9 10:36:25  HUANGXIN create this file, implements the fundemental interface;
-**************************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\dma\dma_regs.h
+ * (C) Copyright 2007-2011
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * huangxin <huangxin@allwinnertech.com>
+ *
+ * some simple description for this code
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 #ifndef _DMA_REGS_
 #define _DMA_REGS_
 
diff --git a/arch/arm/mach-sun4i/include/mach/dma.h b/arch/arm/mach-sun4i/include/mach/dma.h
index 46f503b..66bb96b 100644
--- a/arch/arm/mach-sun4i/include/mach/dma.h
+++ b/arch/arm/mach-sun4i/include/mach/dma.h
@@ -1,20 +1,17 @@
 /*
-**************************************************************************************************************
-*											         eLDK
-*						            the Easy Portable/Player Develop Kits
-*									           desktop system
-*
-*						        	 (c) Copyright 2009-2012, ,HUANGXIN China
-*											 All Rights Reserved
-*
-* File    	: dma.h
-* By      	: HUANGXIN
-* Func		:
-* Version	: v1.0
-* ============================================================================================================
-* 2011-5-9 10:37:24  HUANGXIN create this file, implements the fundemental interface;
-**************************************************************************************************************
-*/
+ * arch\arm\mach-sun4i\include\mach\dma.h
+ * (C) Copyright 2007-2011
+ * Allwinner Technology Co., Ltd. <www.allwinnertech.com>
+ * huangxin <huangxin@allwinnertech.com>
+ *
+ * some simple description for this code
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ */
 #ifndef __ASM_ARCH_DMA_H
 #define __ASM_ARCH_DMA_H __FILE__
 
diff --git a/drivers/media/audio/sun4i_dev_ace.c b/drivers/media/audio/sun4i_dev_ace.c
index bd8435c..e5a38bb 100644
--- a/drivers/media/audio/sun4i_dev_ace.c
+++ b/drivers/media/audio/sun4i_dev_ace.c
@@ -1,12 +1,3 @@
-#include <linux/init.h>
-#include <linux/module.h>
-#include <linux/ioctl.h>
-#include <linux/fs.h>
-#include <linux/device.h>
-#include <linux/err.h>
-#include <linux/list.h>
-#include <linux/errno.h>
-#include <linux/mutex.h>
 /*
  * drivers\media\audio\sun4i_dev_ace.c
  * (C) Copyright 2007-2011
@@ -21,6 +12,16 @@
  * the License, or (at your option) any later version.
  *
  */
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/ioctl.h>
+#include <linux/fs.h>
+#include <linux/device.h>
+#include <linux/err.h>
+#include <linux/list.h>
+#include <linux/errno.h>
+#include <linux/mutex.h>
+
 #include <linux/slab.h>
 #include <linux/preempt.h>
 #include <linux/cdev.h>
-- 
1.8.0

